### PR TITLE
feat(transaction-search): Transaction name search and navigation

### PR DESCRIPTION
**Pull request by udameli**
This PR adds a new type of search result item - `LINK`. The search result item would consist of the transaction name and a link icon. When the icon is clicked, it would guide the user to the transaction summary page. This new search is behind a feature flag and it's necessary for the MEPS E2E testing. I will follow up with the clean up and improved styling.
image

### CODE REVIEW

**Review by k-fish — APPROVED** [review]
Aside from checking about the search changes for that `LINK`, lgtm 👍

**Comment by udameli** [comment]
@getsentry/core-ui I'd really appreciate your feedback on the search item changes here. I just need to ship this new search to unblock meps testing but pls leave comments here and I will address them in a follow up PR. This is behind a feature flag so it should be relatively safe to merge right now

**Pull request merged**
